mp/src/game/server/client.cpp

```diff
@@ -847,52 +848,65 @@ CON_COMMAND( say_team, "Display player message to team" )
 	}
 }
 
-
-//------------------------------------------------------------------------------
-//------------------------------------------------------------------------------
-CON_COMMAND( give, "Give item to player.\n\tArguments: <item_name>" )
+static int WeaponCompletion(const char *pPartial, char commands[COMMAND_COMPLETION_MAXITEMS][COMMAND_COMPLETION_ITEM_LENGTH])
 {
-	CBasePlayer *pPlayer = ToBasePlayer( UTIL_GetCommandClient() ); 
-	if ( pPlayer 
-		&& (gpGlobals->maxClients == 1 || sv_cheats->GetBool()) 
-		&& args.ArgC() >= 2
-		&& !pPlayer->IsObserver() )
+	const auto pCmdName = "give_weapon";
+	char *substring = nullptr;
+	if (Q_strstr(pPartial, pCmdName) && strlen(pPartial) > strlen(pCmdName) + 1)
 	{
-		char item_to_give[ 256 ];
-		Q_strncpy( item_to_give, args[1], sizeof( item_to_give ) );
-		Q_strlower( item_to_give );
+		substring = (char *)pPartial + strlen(pCmdName) + 1;
+	}
 
-		// Don't allow regular users to create point_servercommand entities for the same reason as blocking ent_fire
-		if ( !Q_stricmp( item_to_give, "point_servercommand" ) )
+	int current = 0;
+	// Skip over weapon_none
+	for (int weaponID = WEAPON_PISTOL; weaponID < WEAPON_MAX; weaponID++)
+	{
+		if (!g_pGameModeSystem->GetGameMode()->WeaponIsAllowed((WeaponID_t)weaponID))
+			continue;
+
+	    if ((!substring || !substring[0]) || Q_stristr(g_szWeaponNames[weaponID], substring))
+	    {
+			char command[COMMAND_COMPLETION_ITEM_LENGTH];
+			Q_snprintf(command, COMMAND_COMPLETION_ITEM_LENGTH, "%s %s", pCmdName, g_szWeaponNames[weaponID]);
+	        Q_strncpy(commands[current], command, COMMAND_COMPLETION_ITEM_LENGTH);
+			current++;
+	    }
+	}
+
+	return current;
```

Comment:
Tab/space mixing

Reply:
Resolved in https://github.com/momentum-mod/game/pull/624/commits/79a39fad8fd283aa27e934a67aad6c8806ef7947